File: multiplatform/library/game/src/commonMain/kotlin/cz/johnyapps/mutliplatform/library/game/domain/model/GameBoard.kt
package cz.johnyapps.mutliplatform.library.game.domain.model

data class GameBoard(
    val moves: Set<Move>,
    val size: Size,
) {

    data class Move(
        val column: Int,
        val row: Int,
        val playerId: PlayerId,
    )

    data class Size(
        val columns: Int,
        val rows: Int,
    )
}

File: ios/IosModel/Sources/IosModel/library/game/model/GameBoard.swift
public struct GameBoard : Sendable {
    public let moves: [Move]
    public let size: Size
    
    public init(moves: [Move], size: Size) {
        self.moves = moves
        self.size = size
    }

    public struct Move: Hashable, Sendable {
        public let column: Int
        public let row: Int
        public let playerId: PlayerId
        
        public init(column: Int, row: Int, playerId: PlayerId) {
            self.column = column
            self.row = row
            self.playerId = playerId
        }
    }

    public struct Size : Hashable, Sendable {
        public let columns: Int
        public let rows: Int
        
        public init(columns: Int, rows: Int) {
            self.columns = columns
            self.rows = rows
        }
    }
}
